processor/src/main/resources/org/mapstruct/ap/internal/model/IterableMappingMethod.ftl

```diff
@@ -35,7 +35,7 @@
                 <#if existingInstanceMapping>
                     <#-- we can't clear an existing array, so we've got to clear by setting values to default -->
                     for (int ${index2Name} = 0; ${index2Name} < ${resultName}.length; ${index2Name}++ ) {
-                        ${resultName}[${index2Name}] = ${defaultValue};
+                        ${resultName}[${index2Name}] = ${resultElementType.null};
```

Comment:
I don't think so. Just have a look at the `getDefaultValue()` that was removed. We were doing `getResultElementType().getTypeMirror()...` so if it was `null` we would have had an NPE.